lib/renaissance/auctions/auctions.ex

```diff
@@ -1,50 +1,51 @@
 defmodule Renaissance.Auctions do
   import Ecto.Query
-  alias Renaissance.{Repo, Auction}
+  alias Renaissance.{Auction, Repo, Bids}
+  alias Renaissance.Helpers.{Adapt, Compare}
 
   def create_auction(user_id, params) do
-    details =
-      if Map.has_key?(params, "auction") do
-        params["auction"]
-      else
-        params
-      end
-
-    details =
-      details
-      |> format_price()
+    params =
+      params
+      |> Adapt.format_amount("price")
       |> Map.put("seller_id", user_id)
 
-    Repo.insert(Auction.changeset(%Auction{}, details))
+    Auction.changeset(%Auction{}, params)
+    |> Repo.insert()
   end
 
-  def get(id) do
+  def exists?(id) do
+    Repo.exists?(from a in Auction, where: a.id == ^id)
+  end
+
+  def get!(id) do
     Auction
     |> preload(:seller)
     |> Repo.get!(id)
   end
 
+  def get_seller_id(id) do
+    if exists?(id), do: get!(id).seller_id
```

Comment:
I'm not 100% on the ecto optimization of it, but I'm guessing ecto pulls out the full record with `get!` in which case why have methods in the repo to pull specific things if it's not optimizing the query? Why not have a find_if_exists type call that does get and then in the methods get the attributes?

Reply:
I wouldn't be at all surprised if that's true & can see the value, but I don't know if it would be worthwhile at present. As I see it, at present, this is the only function that retrieved a record, sans preloading, & extracts a specific field without adding additional functionality.